src/util/config.cpp

```diff
@@ -862,7 +862,8 @@ bool configt::set(const cmdlinet &cmdline)
       #endif
 
       // MinGW has extra defines
-      ansi_c.defines.push_back("__int64=\"long long\"");
+      // defines are shell quoted, which gives the wrong result in this case
+      ansi_c.preprocessor_options.push_back("-D__int64=\"long long\"");
```

Comment:
Wouldn't it suffice to just remove the `\"` from the original code as the entire define will be shell-quoted?

Reply:
Your suggestion would create the command line option `-D"__int64=long long"` instead of `-D__int64="long long"`. That is working correctly on a small test case, but the other defines on the command line use the latter format, e.g. `-D__WINT_TYPE__="unsigned int"`. If you are reasonable sure that  `-D"__int64=long long"` will work correctly, I will adapt your suggestion.

Reply:
Do you have a Windows system at hand to try this? I'm pretty sure the `__WINT_TYPE__` case is actually broken. `"` just must not be used on Windows when seeking to do shell quoting.

Reply:
I am trying this on windows and both formats of -D are working correctly. Note that I refer to the format as it appears in the actual system command, i.e. after shell quoting.